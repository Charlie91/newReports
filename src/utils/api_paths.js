
let postfix = PRODUCTION ? '' : '_test'; //пути различаются в девелопменте и продакшене

export let API = {
    main: `https://repo.re-ports.ru/app${postfix}/UserData`,
    auth: `https://repo.re-ports.ru/app${postfix}/Auth`,
    register: `https://repo.re-ports.ru/app${postfix}/Register`,
    nav: `https://repo.re-ports.ru/app${postfix}/Conceptions`,
    cities: `https://repo.re-ports.ru/app${postfix}/Cities?conceptId=`,
    objects: `https://repo.re-ports.ru/app${postfix}/Objects`,
    objectsData: `https://repo.re-ports.ru/app${postfix}/ObjectCache`,
    objectsArrayData: `https://repo.re-ports.ru/app${postfix}/ObjCacheArrays`,
    floors:`https://repo.re-ports.ru/app${postfix}/ObjectFloors?objId=`,
    floorsData:`https://repo.re-ports.ru/app${postfix}/FloorData`,
    innerObjects:`https://repo.re-ports.ru/trc_api${postfix}/shopData?trcId=`,
    userData:`https://repo.re-ports.ru/app${postfix}/UserData`,
    abcAnalysis_shops:`https://repo.re-ports.ru/trc_api${postfix}/abc_xyz_shops`,
    abcAnalysis_categories:`https://repo.re-ports.ru/trc_api${postfix}/abc_xyz_types`,
    imgPath:`https://re-ports.ru/app/img/malls/resize`,
    thumbPath:`https://re-ports.ru/app/img/malls/thumb/`
};