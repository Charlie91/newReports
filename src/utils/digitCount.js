export function digitCount(number) {
    return number.toString().length;
};