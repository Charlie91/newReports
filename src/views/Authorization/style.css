.label-enter {
  margin-top: 20px; }

.label-enter.label-enter-active {
  margin-top: 0;
  transition: margin-top 200ms ease-in; }

.label-leave {
  margin-top: 0; }

.label-leave.label-leave-active {
  margin-top: 20px;
  transition: margin-top 200ms ease-in; }
