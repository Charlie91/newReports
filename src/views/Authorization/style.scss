

//.label-enter {
//  opacity: 0.01;
//}                    // анимация fadeIn/fadeOut
//
//.label-enter.label-enter-active {
//  opacity: 1;
//  transition: opacity 500ms ease-in;
//}
//
//.label-leave {
//  opacity: 1;
//}
//
//.label-leave.label-leave-active {
//  opacity: 0.01;
//  transition: opacity 300ms ease-in;
//}


.label-enter {
  margin-top: 20px;
}

.label-enter.label-enter-active {
  margin-top: 0;
  transition: margin-top 200ms ease-in;
}

.label-leave {
  margin-top:0;
}

.label-leave.label-leave-active {
  margin-top: 20px;
  transition: margin-top 200ms ease-in;
}
