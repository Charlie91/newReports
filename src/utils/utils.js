import {API} from './api_paths';

export function checkEitherLoggedInOrNot(){ //проверка залогинен ли юзер
    let options = {
        method:'GET',
        credentials:'include',
        mode: 'cors'
    };
    return ajaxRequest(API.auth,options)
}

export function ajaxRequest(url,options){   //функция-шаблон для ajax-запросов
    return  fetch(url, options)
        .then(function (response) {
            if (!response.ok) {
                return Promise.reject(new Error(
                    'Response failed: ' + response.status + ' (' + response.statusText + ')'
                ));
            }
            return response.json();
        })
}

export function getCoords(elem) {                         //кроссбраузерная функция получения координат DOM-элемента
    let box = elem.getBoundingClientRect(),
        body = document.body,
        docEl = document.documentElement,
        scrollTop = window.pageYOffset || docEl.scrollTop || body.scrollTop,
        scrollLeft = window.pageXOffset || docEl.scrollLeft || body.scrollLeft,
        clientTop = docEl.clientTop || body.clientTop || 0,
        clientLeft = docEl.clientLeft || body.clientLeft || 0,
        top = box.top + scrollTop - clientTop,
        left = box.left + scrollLeft - clientLeft;

    return {
        top: top,
        left: left
    };
}

export function mobileSidebarHidden() { //скрываем моб-е меню если открыто после перехода по ссылке из меню
    document.body.classList.remove('sidebar-mobile-show');
}

export function formatNumericValue(number){
    if(typeof number === 'number'){
        if(number > 999){
            let str_number = String(number);
            if(str_number.length > 9){
                let millions = str_number.substr(-9,3);
                let billions = str_number.slice(0,str_number.length - 9);
                return billions + ',' +  millions  + ' млрд.'
            }
            if(str_number.length > 6){
                let thousands = str_number.substr(-6,2);
                let millions = str_number.slice(0,str_number.length - 6);
                return millions + ',' +  thousands  + ' млн.'
            }
            if(str_number.length > 3){
                let units = str_number.substr(-3,1);
                let thousands = str_number.slice(0,str_number.length - 3);
                return thousands + ',' + units + ' тыс.'
            }
        }
        return number
    }
    else{
        console.log('Тип передаваемого аргумента - не число')
    }
}