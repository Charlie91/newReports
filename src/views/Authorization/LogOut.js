import React from 'react';
import Authorization from './Authorization.js';
import {ajaxRequest} from './../../utils/utils';
import {Redirect} from 'react-router-dom';
import {API} from './../../utils/api_paths';

export default class LogOut extends Authorization { // Внимание! Этот класс наследует от класса авторизации
    constructor(props){
        super(props);
        this.state={
            isLoggedIn:this.props.isLoggedIn
        }
    }


    logOff(e){  // запрос на выход пользователя\логаут
        e.preventDefault();
        let options = {
            method: 'DELETE',
            credentials: 'include',
            mode: 'cors'
        };
        this.sendDataForLogInAndOut(options);
    }

    componentWillReceiveProps(nextProps){
                this.setState({isLoggedIn:nextProps.isLoggedIn})
    }

    sendDataForLogInAndOut(options){    //обработка ответов на запросы логина\логаута
        ajaxRequest(API.auth,options)
            .then(data => {
                if(data.success){
                    this.setState({isLoggedIn:false})
                }
            })
            .catch(error => console.log(error));
    }
//<button className="btn btn-link" onClick={this.logOff.bind(this)}><i className="fa fa-user-times fa-lg mt-1"></i></button>
    render(){
        return (
            <div>
                { (this.state.isLoggedIn === false) ? <Redirect to="/authorization"/> : ''}
                <button className="settings-btn" onClick={this.logOff.bind(this)}><img src="img/settings.png"/></button>
            </div>
        )
    }

}