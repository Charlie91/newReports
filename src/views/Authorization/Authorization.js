import React, { Component } from 'react';
import './style.scss';
import AuthNav from './../AuthNav/AuthNav';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import { Redirect} from 'react-router-dom';
import {API} from './../../utils/api_paths';
import {ajaxRequest, deleteRegistrationCookies} from './../../utils/utils';


import {
    Row, Col
} from "reactstrap";

export function showDynamicLabel(nameProperty,text){    //динамический показ лейбла у форм
    if(nameProperty){
        return (
            <span className="label-hint"> {text} </span>
        )
    }
}

export function animateDynamicLabel(nameProperty, text){
    return(
        <ReactCSSTransitionGroup
            transitionName="label"
            transitionEnterTimeout={500}
            transitionLeaveTimeout={300}
        >
            {showDynamicLabel(nameProperty,text)}
        </ReactCSSTransitionGroup>
    )
}

export default class Authorization extends Component {


    constructor(props){
        super(props);
        this.state={
            isLoggedIn:null,
            login:'',
            password:''
        }
    }

    setLogin(e){
        this.setState({login:e.target.value})
    }

    setPassword(e){
        this.setState({password:e.target.value})
    }

    validation(){   //валидация полей
        if(!this.state.login || !this.state.password){
            this.setState({hasErrors:true});
            return false;
        }
        else{
            this.setState({hasErrors:false});
            return true;
        }
    }

    getUserData(){  // парсинг данных пользователя после авторизации
        let options = {
            method:'GET',
            credentials:'include',
            mode: 'cors'
        };
        ajaxRequest(API.main,options)
            .then(data => {
                this.setState({
                    userName:data.login
                })
            })
            .catch(error => console.log(error));
    }


    checkEitherLoggedInOrNot(){ //проверка залогинен ли юзер
        let options = {
            method:'GET',
            credentials:'include',
            mode: 'cors'
        };
        ajaxRequest(API.auth ,options)
            .then(data => {
                if (data.authorized === true) {
                    this.setState({isLoggedIn: true});
                    if (("standalone" in window.navigator) && !window.navigator.standalone) {
                        if (this.state.login) localStorage.setItem('login', this.state.login);
                        if (this.state.password) localStorage.setItem('password', this.state.password);
                    }
                }
                else {
                    this.setState({isLoggedIn: false, password: ''});
                    if (("standalone" in window.navigator) && !window.navigator.standalone) {
                        if (localStorage.getItem('login')) this.setState({login: localStorage.getItem('login')});
                        if (localStorage.getItem('password')) this.setState({password: localStorage.getItem('password')});
                    }

                }
            })
            .catch(error => console.log(error));
    }


    logIn(e) {      // запрос на вход\авторизацию пользователя
        if(typeof(e) != "undefined") e.preventDefault();
        if (!this.validation())return;
            let obj = {
                pwd: this.state.password,
                login: this.state.login
            };
            let options = {
                method: 'POST',
                credentials: 'include',
                headers: {
                    "Content-type": "application/json;charset=UTF-8"
                },
                body: JSON.stringify(obj),
                mode: 'cors'
            };
            this.sendDataForLogInAndOut(options);
    }

    logOff(e){  // запрос на выход пользователя\логаут
        e.preventDefault();
        let options = {
            method: 'DELETE',
            credentials: 'include',
            mode: 'cors'
        };
        this.sendDataForLogInAndOut(options);
    }

    sendDataForLogInAndOut(options){    //обработка ответов на запросы логина\логаута
        fetch(API.auth,options)
            .then(function (response) {
                return response.json();
            })
            .then(data => {
                if(data.error) {
                    if (data.error.message === 'Неверный логин или пароль') {
                        this.setState({loginIsIncorrect: true})
                    }
                }
                else
                    this.setState({loginIsIncorrect: false,connectionIsFailed:false});
                this.checkEitherLoggedInOrNot()
            })
            .catch(error => this.setState({connectionIsFailed:true}));
    }


    showError(){        //показ ошибок
        if(this.state.hasErrors)
            return(
                <div className="hintMessage alert alert-danger">введите логин и пароль</div>
            );
        if(this.state.loginIsIncorrect)
            return(
                <div className="hintMessage alert alert-danger">логин и пароль неверны</div>
            );
        if(this.state.connectionIsFailed)
            return(
                <div className="hintMessage alert alert-danger">Соединение потеряно</div>
            )
    }
//                                    autoComplete="new-password"                                     size="5"


    showForm(){
            return(
                <div>
                    <div>{(("standalone" in window.navigator) && !window.navigator.standalone) ? 'standalone' : 'nope' }</div>
                    <form action="#" autoComplete="off" method="POST">
                        <div className="form-group">
                            <label>
                                {animateDynamicLabel(this.state.login, 'Логин')}
                                <input
                                    type="text"
                                    value={this.state.login}
                                    onChange={this.setLogin.bind(this)}
                                    autoComplete="on"
                                    className="form-control"
                                    placeholder="Логин"
                                />
                            </label>
                            <label>
                                {animateDynamicLabel(this.state.password, 'Пароль')}
                                <input
                                    type="password"
                                    value={this.state.password}
                                    onChange={this.setPassword.bind(this)}
                                    autoComplete="on"
                                    className="form-control"
                                    placeholder="Пароль"
                                />
                            </label>
                            {this.showError()}
                        </div>
                        <Row>
                            <Col xs={{size: 8, offset:2}} md={{ size: 4, offset:4 }}>
                                <button
                                    type="submit"
                                    className="btn auth-btn"
                                    onClick={this.logIn.bind(this)}
                                    disabled={!(this.state.login && this.state.password)}
                                >
                                    Войти
                                </button>
                            </Col>
                        </Row>
                    </form>
                </div>
            )
    }

    componentDidMount(){
        this.checkEitherLoggedInOrNot();//при заходе на стр сразу проверяем авторизован ли юзер
    }

    render() {
        return (
            <div className="auth-window animated fadeIn">
                <AuthNav/>
                {
                    (this.state.isLoggedIn) ?
                        <Redirect to="/dashboard"/>
                        :
                        this.showForm()
                }
                <div className="blur_layout"> </div>
            </div>
        )
    }
}


