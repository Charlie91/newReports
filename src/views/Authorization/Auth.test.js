import fetch from 'node-fetch';
import Authorization from './Authorization';
import renderer from 'react-test-renderer';
import React from 'react';
import * as utils from './../../utils/utils';


jest.mock('react-router-dom');
jest.mock('../AuthNav/AuthNav', () => 'AuthNav');

utils.ajaxRequest = jest.fn(() => new Promise(function(resolve, reject) {
    resolve('Promise resolved');//мокаем ajax request
}));

utils.insertFacebookSDK = jest.fn(() => 'facebook sdk inserting');//мокаем скрипт ФБ


test('logged in succesfully', () => {   //тестирование авторизации
    let user = {        //данные тестового пользователя
        login: 'Charlie91',
        pwd: '123456'
    };
    let options = {
        method: 'POST',
        credentials: 'include',
        headers: {
            "Content-type": "application/json;charset=UTF-8"
        },
        body: JSON.stringify(user),
        mode: 'cors'
    };
    fetch('https://repo.re-ports.ru/app_test/Auth',options)
        .then(response => {
            return response.json();
        })
        .then(data => expect(data).toMatchObject({
            success:true
        }))
        .catch(err => console.log(err))
});


describe('Authorization works correctly', () => {//тестирование класса
    it('Auth component renders correctly', () => {
        const tree = renderer
            .create(<Authorization/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
