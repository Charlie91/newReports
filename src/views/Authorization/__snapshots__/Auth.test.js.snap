// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorization works correctly Auth component renders correctly 1`] = `
<div
  className="auth-window animated fadeIn"
>
  <AuthNav />
  <div>
    <form
      action="#"
      autoComplete="on"
      className="auth-form"
      method="POST"
    >
      <div
        className="form-group"
      >
        <label>
          <span />
          <input
            autoComplete="on"
            className="form-control"
            onChange={[Function]}
            placeholder="Логин"
            type="text"
            value=""
          />
        </label>
        <label>
          <span />
          <input
            autoComplete="on"
            className="form-control pass"
            onChange={[Function]}
            placeholder="Пароль"
            type="password"
            value=""
          />
        </label>
      </div>
      <div
        className="row"
      >
        <div
          className="col-8 offset-2 col-md-4 offset-md-4"
        >
          <button
            className="btn auth-btn"
            disabled={true}
            onClick={[Function]}
            type="submit"
          >
            Войти
          </button>
          <a
            href="#"
            onClick={[Function]}
          >
            Login with Facebook
          </a>
        </div>
      </div>
    </form>
  </div>
  <div
    className="blur_layout"
  >
     
  </div>
</div>
`;
